### PR TITLE
feat(amazonq): adding support to accept java 17 projects

### PR DESCRIPTION
**Pull request by ntarakad-aws**
## Problem

Currently /transform feature can only accept java 8/11 projects.

## Solution

Updating user form to allow jdk 17 as source version.

### CODE REVIEW

File: packages/amazonq/.changes/next-release/Feature-76e907b5-f09d-4b24-bc60-156471d3cb22.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Feature",
+	"description": "Amazon Q Code Transformation feature: allow Java 17 projects to be transformed"
```
Comment (hayemaxi) [inline packages/amazonq/.changes/next-release/Feature-76e907b5-f09d-4b24-bc60-156471d3cb22.json:3]:
```suggestion
	"description": "Code Transform: Enable support for Java 17 projects."
```

File: packages/core/src/codewhisperer/models/constants.ts [hunk 2]
```diff
@@ -637,6 +637,9 @@ export const macJava8HomeHelpChatMessage =
 export const macJava11HomeHelpChatMessage =
     'To find the JDK path, run the following command in a new terminal:  `/usr/libexec/java_home -v 11`'
 
+export const macJava17HomeHelpChatMessage =
+    'To find the JDK path, run the following command in a new terminal:  `/usr/libexec/java_home -v 17`'
```
Comment (hayemaxi) [inline packages/core/src/codewhisperer/models/constants.ts:641]: Looks like the same string is duplicated 3 times, with only a 2-3 character difference. Can you use a templated string instead and pass the version number?

Comment (dhasani23) [reply packages/core/src/codewhisperer/models/constants.ts:641]: @ntarakad-aws example of this: https://github.com/aws/aws-toolkit-vscode/blob/e8d62f1576a255ea105c4b7ba0b17b54cb9a87cb/packages/core/src/codewhisperer/models/constants.ts#L496

**Comment by dhasani23** [comment]
Waiting for E2E testing before approving. We should make sure to test when `JAVA_HOME` does and does not point to a valid JDK17 path.

File: packages/core/src/amazonq/webview/ui/quickActions/generator.ts [hunk 3]
```diff
@@ -40,7 +40,7 @@ export class QuickActionGenerator {
                         ? [
                               {
                                   command: '/transform',
-                                  description: 'Transform your Java 8 or 11 Maven project to Java 17',
+                                  description: 'Transform your Java 8, Java 11, or Java 17 Maven projects ',
```
Comment (dhasani23) [inline packages/core/src/amazonq/webview/ui/quickActions/generator.ts:43]: `Transform you Java project`

File: packages/core/src/amazonq/webview/ui/tabs/constants.ts [hunk 4]
```diff
@@ -35,6 +35,6 @@ What would you like to work on?`,
         placeholder: 'Open a new tab to chat with Q',
         welcome: `Welcome to Code Transformation!
 
-I can help you upgrade your Java 8 and 11 codebases to Java 17.`,
+I can help you upgrade your Java 8 and Java 11 codebases to Java 17 and upgrade Java 17 code with up to date libraries and other dependencies.`,
```
Comment (dhasani23) [inline packages/core/src/amazonq/webview/ui/tabs/constants.ts:38]: Delete and just do `Welcome to Code Transformation!`

**Review by dhasani23 — APPROVED** [review]

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**